### ISSUE TITLE
Don't honor .gitignore when searching for services

### ISSUE THREAD

**Issue by tristanpemble** [datawire/forge#106]
example directory tree:

```
.
├── .gitignore
├── README.md
├── forge.yaml
├── k8s
│   └── .gitkeep
├── service.yaml
└── workspace
    ├── .gitignore
    └── customer-portal
        ├── Dockerfile
        ├── k8s
        │   └── deployment.yaml
        └── service.yaml
```

contents of `.gitignore`s:

```
λ cat .gitignore
forge.yaml
.forge

λ cat workspace/.gitignore
*
!.gitignore
```

`forge pull`:

```
λ forge pull
║ [.forge/payment-processor] git pull
║ Already up-to-date.
║ [.forge/customer-portal] git pull
║ Already up-to-date.
║ git pull
║ Already up-to-date.
║ [.forge/message-broker] git pull
║ Already up-to-date.
║ [.forge/admin-portal] git pull
║ Already up-to-date.
║ 43 tasks run, 0 errors
```

notice that it is pulling in `.forge/customer-portal` instead of `workspace/customer-portal`

**Comment by tristanpemble** [comment]
I created https://github.com/forge-playground/development-environment to sort of illustrate how I am trying to use forge, maybe it will help

**Comment by rhs** [comment]
What about having an explicit search path in forge.yaml for services? I'm just thinking if we ignore .gitignore when searching for services, then we have no way to avoid searching some potentially large directories, e.g. node_modules.

**Comment by tristanpemble** [comment]
works for me 👍

**Comment by rhs** [comment]
This should be fixed in 0.3.21. I added a search-path attribute to forge.yaml that allows you to opt into a search path.
